pantavisor.c

```diff
@@ -872,25 +874,34 @@ static pv_state_t pv_shutdown(struct pantavisor *pv, shutdown_type_t t)
 	pv_state_stop_force(pv->state);
 	ph_logger_stop_force();
 
-	// at this point, we can shutdown if not in appengine
-	if (pv_config_get_system_init_mode() != IM_APPENGINE)
-		reboot(shutdown_type_reboot_cmd(t));
+	// unmount disks
+	pv_disks_umount_all(pv->state);
 
 	// close pvctrl
 	pv_ctrl_socket_close(pv->ctrl_fd);
 
-	// stop logs now
-	pv_logserver_stop();
+	// stop all logs but stdout
+	pv_logserver_degrade();
 	pv_log_umount();
 
-	// unmount everything
 	pv_storage_umount();
 	pv_mount_umount();
 	pv_init_umount();
 
 	// free up memory
 	pv_bootloader_remove();
-	pv_remove(pv);
+
+	// at this point, we can shutdown if not in appengine
+	if (initmode != IM_APPENGINE) {
+		pv_log(INFO, "shutdown complete, rebooting in 2 second ...");
+		sleep(2);
```

Comment:
out of curiosity, why do we need the sleep here? Also, @fortizc is it worth it to call pv_logserver_stop in this case too?

Reply:
we can call stop also in system reboot case, but we want to see even the last of messages that we can retrieve like a dmesg output at the end telling us that it will reboot etc.... so i thought keeping a single process that has no resources lockd to disk resources should be fine.